docker/patch/latest/megatron.patch

```diff
@@ -12,38 +61,16 @@ index 41c21d93d..ef80f72d6 100644
              err_msg = f'Common file {load_path} does not exist'
              if MultiStorageClientFeature.is_enabled():
 diff --git a/megatron/core/dist_checkpointing/strategies/torch.py b/megatron/core/dist_checkpointing/strategies/torch.py
-index ccf5242a2..9b6d3e31f 100644
+index 5a1ea308d..87826e4df 100644
 --- a/megatron/core/dist_checkpointing/strategies/torch.py
 +++ b/megatron/core/dist_checkpointing/strategies/torch.py
-@@ -427,6 +427,15 @@ def _restore_dict_types(x: Union[dict, list, Any], keys_template: Union[dict, li
-             _restore_dict_types(x_val, templ_val)
- 
- 
-+@dataclass
-+class MCoreMetadata(Metadata):
```

Comment:
for those surprised to see "why would we see any lines deleted when we rebase our patch onto a newer Megatron": I guess what happened here is that the MCoreMetadata dataclass got upstreamed? Megatron v0.15.0 already has this, so there's no longer a need for the patch to add it:

https://github.com/NVIDIA/Megatron-LM/blob/0112156741214c833a47d9f92911400ae96d724d/megatron/core/dist_checkpointing/strategies/torch.py#L462